src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Utf8Formatter/FormattingHelpers.cs

```diff
@@ -112,41 +113,40 @@ public static void WriteDigits(uint value, Span<byte> buffer)
         /// This method performs best when the starting index is a constant literal.
         /// </summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static void WriteFourDecimalDigits(uint value, Span<byte> buffer, int startingIndex = 0)
+        public static unsafe void WriteFourDecimalDigits(uint value, Span<byte> buffer, int startingIndex = 0)
         {
             Debug.Assert(value <= 9999);
+            Debug.Assert(startingIndex <= buffer.Length - 4);
```

Comment:
startingIndex is not checked for negative values... and the JIT doesn't do bound checks here anymore... 

is that intended (e.g. can never happen) or an oversight? same below...

Reply:
We could change the assert to flag negatives as well, but it can't happen: all of the calls to this method use a constant non-negative startingIndex.